experimental/Schemes/Sheaves.jl

```diff
@@ -0,0 +1,105 @@
+export AbsSheaf
+export space, restriction_map
+export SheafOnScheme
+export RingOfRegularFunctions
+
+########################################################################
+# The AbsSheaf interface                                               #
+########################################################################
+@Markdown.doc """
+    space(F::AbsSheaf) 
+
+For a sheaf ``ℱ`` on a space ``X`` return ``X``.
+"""
+function space(F::AbsSheaf) 
+  return space(underlying_sheaf(F))
+end
+
+@Markdown.doc """
+    (F::AbsSheaf{<:Any, OpenType})(U::T) where {T<:OpenType}
+
+For a sheaf ``ℱ`` on a space ``X`` and an (admissible) open set 
+``U ⊂ X`` check whether ``U`` is open in ``X`` and return ``ℱ(U)``.
+"""
+function (F::AbsSheaf{<:Any, OpenType, OutputType})(U::T) where {OpenType, OutputType, T<:OpenType}
+  return (underlying_sheaf(F))(U)::OutputType
+end
```

Comment:
I would allow to turn of caching. Sometimes serious computations run out of memory because of too much caching. 
Also here and elsewhere the docstrings can be simplified. I think it is not helpful to display all the types there. 
It is more important to grasp at a first glance what is going on. 
The programmer can just look at the signature.